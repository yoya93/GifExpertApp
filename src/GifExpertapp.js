import React, { useState } from "react";

import "./index.css";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertapp = ({ defaultscategories = [] }) => {
  const [categories, setCategories] = useState(defaultscategories);

  return (
    <>
      <h1>GifExpertapp </h1>

      <AddCategory setCategories={setCategories} />

      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
