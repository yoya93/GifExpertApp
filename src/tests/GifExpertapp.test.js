import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { GifExpertapp } from "../GifExpertapp";

describe("Pruebas en <GifExpertapp /> ", () => {
  test("debe mostrarse correctamente ", () => {
    const wrapper = shallow(<GifExpertapp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe renderizar un numero de items por ctegories recibidas", () => {
    const categories = ["Griezman", "Messi"];
    const wrapper = shallow(<GifExpertapp defaultscategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
