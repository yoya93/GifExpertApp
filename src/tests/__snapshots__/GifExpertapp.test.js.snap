// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertapp />  debe mostrarse correctamente  1`] = `
<Fragment>
  <h1>
    GifExpertapp 
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertapp />  debe renderizar un numero de items por ctegories recibidas 1`] = `
<Fragment>
  <h1>
    GifExpertapp 
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Griezman"
      key="Griezman"
    />
    <GifGrid
      category="Messi"
      key="Messi"
    />
  </ol>
</Fragment>
`;
